03_Modules/Transactions/src/module/services/transaction.event.service.ts

```diff
@@ -0,0 +1,194 @@
+        transactionEventRequest.idToken as IdTokenType
+      );
+
+    // Transaction in this TransactionEventRequest has already been saved, so there should only be 1 active transaction for idToken
+    if (activeTransactions!.length > 1) {
+      transactionEventResponse.idTokenInfo = {
+        status: AuthorizationStatusEnumType.ConcurrentTx,
+        // groupIdToken: transactionEventResponse.idTokenInfo?.groupIdToken, // todo line seems to not do anything
+        // TODO determine how/if to set personalMessage
+      };
+    }
+  }
+
+  private async authorizeAndGetTransactionEventResponse(
+    transactionEvent: TransactionEventRequest
+  ) {
+    const authorization = await this.authorizeRepository?.readByQuery({
+      idToken: transactionEvent.idToken?.idToken as string,
+      type: transactionEvent.idToken?.type as IdTokenEnumType,
+    });
+    const transactionEventResponse: TransactionEventResponse = {
+      idTokenInfo: {
+        status: AuthorizationStatusEnumType.Unknown,
+        // TODO determine how/if to set personalMessage
+      },
+    };
+    if (authorization) {
+      if (authorization.idTokenInfo) {
+        // Extract DTO fields from sequelize Model<any, any> objects
+        const idTokenInfo =
+          this.transactionEventMapper?.mapAuthorizationToIdTokenInfo(
+            authorization
+          )!;
+        this.setTransactionEventResponseTokenInfo(
+          transactionEventResponse,
+          idTokenInfo
+        );
+      } else {
+        // Assumed to always be valid without IdTokenInfo
+        transactionEventResponse.idTokenInfo = {
+          status: AuthorizationStatusEnumType.Accepted,
+          // TODO determine how/if to set personalMessage
+        };
+      }
+    }
+    return transactionEventResponse;
+  }
+
+  private setTransactionEventResponseTokenInfo(
+    transactionEventResponse: TransactionEventResponse,
+    idTokenInfo: IdTokenInfoType
+  ) {
+    if (idTokenInfo.status == AuthorizationStatusEnumType.Accepted) {
+      if (
+        idTokenInfo.cacheExpiryDateTime &&
+        new Date() > new Date(idTokenInfo.cacheExpiryDateTime)
+      ) {
+        transactionEventResponse.idTokenInfo = {
+          status: AuthorizationStatusEnumType.Invalid,
+          groupIdToken: idTokenInfo.groupIdToken,
+          // TODO determine how/if to set personalMessage
+        };
+      } else {
+        // TODO: Determine how to check for NotAllowedTypeEVSE, NotAtThisLocation, NotAtThisTime, NoCredit
+        // TODO: allow for a 'real time auth' type call to fetch token status.
+        transactionEventResponse.idTokenInfo = idTokenInfo;
+      }
+    } else {
+      // IdTokenInfo.status is one of Blocked, Expired, Invalid, NoCredit
+      // N.B. Other non-Accepted statuses should not be allowed to be stored.
+      transactionEventResponse.idTokenInfo = idTokenInfo;
+    }
+  }
+
+  public sendCallResultWithMessage(
```

Comment:
`sendCallResultWithMessage` has been defined in BaseModule.  Is there any better way to avoid this duplication?

Reply:
this is a temporary workaround because in this case this is the `TransactionEventService` and it does not extend the BaseModule, while it is needed for the `sendTransactionEventConfirmation` method - but I can see a base service, or other ways of injecting the module to get this functionality across implemented in the near future for sure! I think with the introduction of dependency injection, we have a lot of flexibility in re-organizing how things are being intialized, and so this
```
return this.sender?.sendResponse(message, payload)!;
```
will potentially be moved elsewhere!